Filter/ModelAutocompleteFilter.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/*
+ * This file is part of the Sonata package.
+ *
+ * (c) Thomas Rabaix <josluis.lopes@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DoctrineMongoDBAdminBundle\Filter;
+
+use Sonata\CoreBundle\Form\Type\EqualType;
+use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
+use Doctrine\Common\Collections\Collection;
+
+class ModelAutocompleteFilter extends Filter
+{
+
+    public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
+    {
+        if (!$data || !is_array($data) || !array_key_exists('value', $data)) {
+            return;
+        }
+
+        if ($data['value'] instanceof Collection) {
+            $data['value'] = $data['value']->toArray();
+        }
+
+        $field = $this->getIdentifierField($field);
+
+        if (is_array($data['value'])) {
+            $this->handleMultiple($queryBuilder, $alias, $field, $data);
+        } else {
+            $this->handleScalar($queryBuilder, $alias, $field, $data);
+        }
+
+    }
+
+    /**
+     *
+     * @param  ProxyQueryInterface $queryBuilder
+     * @param  type                $alias
+     * @param  type                $field
+     * @param  type                $data
+     * @return type
+     */
+    protected function handleMultiple(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
+    {
+        if (count($data['value']) == 0) {
```

Comment:
Always use strict operators (`===`) and yoda style (`0 === count($data['value']))`.